homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,243 @@
+                        title=f"{entry.title} - {entry.unique_id}",
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+            return ret
+        identifier_parts = item.identifier.split("/")
+        diskstation: SynologyDSMData = self.hass.data[DOMAIN][identifier_parts[0]]
+
+        if len(identifier_parts) == 1:
+            # Get Albums
+            # The library works sync, and this expects async calls
+            try:
+                albums = await diskstation.api.photos.get_albums()
+            except SynologyDSMException:
+                return []
+
+            ret = []
+            ret.append(
+                BrowseMediaSource(
+                    domain=DOMAIN,
+                    identifier=f"{item.identifier}/0",
+                    media_class=MediaClass.DIRECTORY,
+                    media_content_type=MediaClass.IMAGE,
+                    title="All images",
+                    can_play=False,
+                    can_expand=True,
+                )
+            )
+            for album in albums:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{item.identifier}/{album.album_id}",
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=album.name,
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+
+            return ret
+
+        # Request items of album
+        # Get Items
+        # The library works sync, and this expects async calls
+        album = SynoPhotosAlbum
+        album.album_id = int(identifier_parts[1])
+        try:
+            items = await diskstation.api.photos.get_items_from_album(album, 0, 1000)
+        except SynologyDSMException:
+            return []
+
+        ret = []
+        for items_item in items:
+            mime_type, _ = mimetypes.guess_type(items_item.file_name)
+            assert isinstance(mime_type, str)
+            if mime_type.startswith("image/"):
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{identifier_parts[0]}/{identifier_parts[1]}/{items_item.thumbnail_cache_key}/{items_item.file_name}",
+                        media_class=MediaClass.IMAGE,
+                        media_content_type=mime_type,
+                        title=items_item.file_name,
+                        can_play=True,
+                        can_expand=False,
+                        thumbnail=await self.async_get_thumbnail(
+                            items_item.thumbnail_cache_key,
+                            items_item.item_id,
+                            "sm",
+                            identifier_parts[0],
```

Comment:
```suggestion
                            album_item,
                            diskstation,
```

use the objects as is

Reply:
I still had to change the file size manually to sm. Because otherwise it would be xl. And that would be way too large for thumbnails